pkg/controller/servicemesh/controlplane/controller.go

```diff
@@ -53,36 +64,61 @@ func newReconciler(mgr manager.Manager, operatorNamespace string) reconcile.Reco
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	// Create a new controller
-	c, err := controller.New("servicemeshcontrolplane-controller", mgr, controller.Options{Reconciler: r})
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
 	if err != nil {
 		return err
 	}
 
 	// Watch for changes to primary resource ServiceMeshControlPlane
-	// XXX: hack: remove predicate once old installation mechanism is removed
 	err = c.Watch(&source.Kind{Type: &v1.ServiceMeshControlPlane{}}, &handler.EnqueueRequestForObject{})
 	if err != nil {
 		return err
 	}
 
-	// XXX: consider adding watches on created resources.  This would need to be
-	// done in the reconciler, although I suppose we could hard code known types
-	// (ServiceAccount, Service, ClusterRole, ClusterRoleBinding, Deployment,
-	// ConfigMap, ValidatingWebhook, MutatingWebhook, MeshPolicy, DestinationRule,
-	// Gateway, PodDisruptionBudget, HorizontalPodAutoscaler, Ingress, Route).
-	// TODO(user): Modify this to be the types you create that are owned by the primary resource
-	// Watch for changes to secondary resource Pods and requeue the owner ControlPlane
-	// err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
-	// 	IsController: true,
-	// 	OwnerType:    &v1.ControlPlane{},
-	// })
-	// if err != nil {
-	// 	return err
-	// }
+	// watch created resources for use in synchronizing ready status
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, ownedResourceEventMapper, ownedResourcePredicates)
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &appsv1.StatefulSet{}}, ownedResourceEventMapper, ownedResourcePredicates)
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &appsv1.DaemonSet{}}, ownedResourceEventMapper, ownedResourcePredicates)
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
+var ownedResourceEventMapper = &handler.EnqueueRequestsFromMapFunc{
+	ToRequests: handler.ToRequestsFunc(func(ownedResource handler.MapObject) []reconcile.Request {
+		owner := metav1.GetControllerOf(ownedResource.Meta)
+		if owner == nil || owner.Kind != "ServiceMeshControlPlane" || owner.APIVersion != v1.SchemeGroupVersion.String() {
+			return nil
+		}
+		return []reconcile.Request{
+			reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name: owner.Name, Namespace: ownedResource.Meta.GetNamespace(),
+				},
+			},
+		}
+	}),
+}
```

Comment:
Doesn't the existing `handler.EnqueueRequestForOwner` do the same thing?

Reply:
We're mapping from a `Deployment` event to a `ServiceMeshControlPlane` event, so this is similar to what we're doing in the `ServiceMeshMemberRoll` controller for namespaces..

That said, it looks like there's an 'EnqueueRequestForOwner` that would do the trick.

Reply:
Updated to use `EnqueueRequestForOwner`.